addons/pos_loyalty/static/src/js/ProductScreen.js

```diff
@@ -56,7 +56,20 @@ export const PosLoyaltyProductScreen = (ProductScreen) =>
                     return false;
                 }
                 const trimmedCode = code.trim();
-                if (trimmedCode && trimmedCode.startsWith('044')) {
+                const barcodeParser = this.env.barcode_reader.barcode_parser || this.env.barcode_reader.fallbackBarcodeParser;
```

Comment:
The rules might exist in either `barcode_parser` or `fallbackBarcodeParser`. To ensure comprehensive coverage, you can merge these rules. Keep in mind that `fallbackBarcodeParser` may be undefined.